source/slang/check.cpp

```diff
@@ -2249,18 +2251,18 @@ namespace Slang
             // check the parameters
             for (auto m : genericDecl->Members)
             {
-                if (auto typeParam = m.As<GenericTypeParamDecl>())
+                if (auto typeParam = as<GenericTypeParamDecl>(m))
```

Comment:
The switch from `m.As<...>()` to `as<...>(m)` here kind of surprised me, because it drops the `RefPtr`-ness. Was this something where you considered how things were being used and determined that it was safe to not have the `auto` local be a `RefPtr`?